src/pipapo/utils/domain.py

```diff
@@ -0,0 +1,150 @@
+"""Geometry utils."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Protocol
+
+import numpy as np
+
+from pipapo.utils import sphere_plane_intersection
+from pipapo.utils.interfaces import InterfaceContainer
+from pipapo.utils.type_hinting import float_np_array
+
+if TYPE_CHECKING:
+    from pipapo.particles import ParticleContainer
+
+
+# pylint: disable=too-few-public-methods
+class Domain(Protocol):
+    """Domain geometry."""
+
+    def get_particles_in_domain(
+        self,
+        particles: ParticleContainer,
+    ) -> ParticleContainer:
+        """Get particles inside the domain.
+
+        Args:
+            particles: Particles to check for inclusion
+
+        Returns:
+            Particles inside the domain
+        """
+
+
+@dataclass
+class Box(Domain):
+    """Axis-aligned box.
+
+    Attributes:
+        center: Center of the box
+        lengths: Lengths of the box edges
+    """
+
+    center: float_np_array
+    lengths: float_np_array
+
+    def get_particles_in_domain(
+        self, particles: ParticleContainer[Any]
+    ) -> ParticleContainer[Any]:
+        """Get particles inside the box.
+
+        Args:
+            particles: Particles to check for inclusion
+
+        Returns:
+            Particles inside the box
+        """
+        in_box_mask = np.all(
+            np.abs(particles.position - self.center) <= 0.5 * self.lengths,
+            axis=1,
+        )
+
+        return particles[np.arange(len(particles))[in_box_mask]]
+
+
+# Boundaries
+class Boundary(Protocol):
+    """Boundary geometry."""
+
+    def get_interfaces(self, particles: ParticleContainer) -> InterfaceContainer:
+        """Get particle boundary pairs.
+
+        Args:
+            particles: Particles to check for contact
+
+        Returns:
+            particle wall pairs
```

Comment:
Like for the Plane below?
```suggestion
            Particle boundary pairs
```